mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -2391,12 +2391,28 @@ mlir::vector::isBroadcastableTo(Type srcType, VectorType dstVectorType,
   // (all leading dimensions are simply duplicated).
   int64_t lead = dstRank - srcRank;
   for (int64_t r = 0; r < srcRank; ++r) {
+    bool mismatch = false;
+
+    // Check fixed-width dims
     int64_t srcDim = srcVectorType.getDimSize(r);
     int64_t dstDim = dstVectorType.getDimSize(lead + r);
-    if (srcDim != 1 && srcDim != dstDim) {
+    if ((srcDim != 1 && srcDim != dstDim))
+      mismatch = true;
+
+    // Check scalable flags
+    bool srcDimScalableFlag = srcVectorType.getScalableDims()[r];
+    bool dstDimScalableFlag = dstVectorType.getScalableDims()[lead + r];
+    if ((srcDim == 1 && srcDimScalableFlag && dstDim != 1) ||
+        (srcDimScalableFlag && !dstDimScalableFlag))
```

Comment:
It got me thinking, what would be the expected behaviour of something like:
```
 %0 = vector.broadcast %arg0 : vector<nxf32> to vector<[n]xf32>
```
IMO it should not be supported as physically equivalent to a usecase
```
%1 = vector.broadcast %arg0 : vector<nxf32> to vector<vscale*nxf32>
```
Which is not invalid for fixed dimensions. Do you think this handles the cases ?
```suggestion
        (srcDimScalableFlag != dstDimScalableFlag))
```

Reply:
If you have e.g. `[2]` and `[4]` (i.e. `vscale * 2` and `vscale * 4`), then that's already "rejected" as "mismatching dims":
* https://github.com/llvm/llvm-project/blob/0dcada94bb1ae79f0edd91013038098c62a96b3b/mlir/lib/Dialect/Vector/IR/VectorOps.cpp#L2396

Is that the case you had in mind?

Reply:
The case I pointed out was more src = `2`  and dest = `[2]`. srcDim == dstDim, so no mismatch on line 2399.  and we have `!srcDimScalableFlag`  so no mismatch on line 2406. Whereas I think this is wrong.
```
 %0 = vector.broadcast %arg0 : vector<2xf32> to vector<[2]xf32>
```

Reply:
Ah, nice, great catch! In my head I had one case that wouldn't work with `!=`, but now I am failing to recall that 😂 

Let me send an update - thanks very much for pointing this out 🙏🏻